tools/cockpit.spec

```diff
@@ -65,6 +65,11 @@ ExcludeArch: %{ix86}
 %endif
 %endif
 
+%define enable_multihost 1
+%if 0%{?rhel} >= 10
+%define enable_multihost 0
+%endif
```

Comment:
This is missing for Fedora rawhide.

It would also need the corresponding changes to the arch and Debian packaging scripts.

That said, this is the wrong way around IMHO: The default should be that multihost is *disabled*, and  the configure option can then turn it on by default for the stable OSes -- then that list shrinks to zero over time.

Reply:
> That said, this is the wrong way around IMHO: The default should be that multihost is _disabled_, and the configure option can then turn it on by default for the stable OSes -- then that list shrinks to zero over time.

Yeah... I did this on purpose, tbh.  I am not convinced yet that our immediate goal should be to switch off host switching by default everywhere.

There is nothing OS specific about this. Multi hosts are not secure on one OS, but problematic on others. The motivation to make this OS specific is to not upset users with a sudden, unwelcome change. And there is no external factor that makes this more urgent now, as compared to half a year ago. So, no rush, imo.

Going forward, I think we should put effort into educating people about the implications, inside Cockpit, with links to a dedicated web page / blog post. That page can explain the alternatives (cockpit-client, container tabs, ...), and also explain the config setting.

We can/should encourage people to switch the switcher off, but I think we shouldn't just do it for them.


I know this isn't your plan, and I am happy to disable the host switcher here on more OSes. But maybe the above is convincing.

Reply:
We can agree to disagree here. It seems wildly arbitrary to me to *just* disable this on RHEL 10, where we get about zero field testing and feedback. As you say this problem totally isn't RHEL specific. I'm happy to be overridden here given that you do the work -- but then this should be justified explicitly in the commit message.

Note that like this it will also be enabled for RHEL 11, aka. Fedora ELN.

Reply:
> And there is no external factor that makes this more urgent now, as compared to half a year ago. So, no rush, imo.

If that's your stanza, then frankly we should just declare the whole thing as "wontfix" and not disable it anywhere. Instead it should be documented/warned about more strongly.

Reply:
> > And there is no external factor that makes this more urgent now, as compared to half a year ago. So, no rush, imo.
> 
> If that's your stance, then frankly we should just declare the whole thing as "wontfix" and not disable it anywhere. Instead it should be documented/warned about more strongly.

Yes, I would be happy with that.